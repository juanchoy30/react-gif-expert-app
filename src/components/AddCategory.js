import React, { useState } from 'react'

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Hello World');

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('submit done')
    };

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange } 
            />
        </form>
    );
}

export default AddCategory;