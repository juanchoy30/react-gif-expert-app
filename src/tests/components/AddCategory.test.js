import React from 'react';
import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"

describe('Component tests', () => {

    const setCategories = () => {};

    test('should render correctly', () => {
        
        const wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
        expect( wrapper ).toMatchSnapshot();


    })
    
})
