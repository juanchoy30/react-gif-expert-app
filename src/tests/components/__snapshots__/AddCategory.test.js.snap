// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
